src/config-file-schema.ts

```diff
@@ -126,8 +124,17 @@ export function ConfigFileSchema<T>(opts: {
     try {
       const parsed = parse(serialized);
       return parsed;
-    } catch (error) {
-      return undefined;
+    } catch (ex: any) {
```

Comment:
Does it make more sense to do this error handling in `parse()` I'm trying to understand what the user would expect if they set values for JSONDecodeError. I think they would expect it to be effective no matter which function they called.

Reply:
Oh, I see that parse isn't actually part of the "object" here. I think this looks good!

Reply:
Well, actually on second thought, one thing I've thought about is enabling different parsing functions, YAML for example. Does it make more sense to capture all parsing errors and treat them the same? For example, on line 39 call it `parseError` and catch all parsing errors in `readParsed()`. That would enable the user to generically handle parsing errors in their application.

Reply:
Ahh yeah that makes sense! OK let me make that change real quick and see if I understood correctly. Thanks!